minimesos/src/main/java/com/containersol/minimesos/docker/DockerClientFactory.java

```diff
@@ -3,29 +3,77 @@
 import com.github.dockerjava.api.DockerClient;
 import com.github.dockerjava.core.DockerClientBuilder;
 import com.github.dockerjava.core.DockerClientConfig;
-import org.apache.commons.lang.StringUtils;
+import java.io.File;
+import java.util.Properties;
+
+import static com.github.dockerjava.core.DockerClientConfig.*;
 
 /**
  * Factory for creating {@link DockerClient}s
  */
 public class DockerClientFactory {
-
     private static DockerClient dockerClient;
+    private static DockerClientConfigBuilder configBuilder = new DockerClientConfigBuilder();
+    private static String dockerHostEnv;
+    private static String dockerCertPathEnv;
+    private static String dockerHostPrp;
+    private static String dockerCertPathPrp;
+    private static String dockerUri;
+    private static String dockerCertPath;
+    private static Properties properties = System.getProperties();
 
     public static DockerClient build() {
-        if (dockerClient == null) {
-            DockerClientConfig.DockerClientConfigBuilder builder = DockerClientConfig.createDefaultConfigBuilder();
-            builder.withVersion("");
+        DockerClientConfig config = populateConfigBuilder(configBuilder).build();
+        dockerClient = DockerClientBuilder.getInstance(config).build();
+        return dockerClient;
+    }
 
-            String dockerHostEnv = System.getenv("DOCKER_HOST");
-            if (StringUtils.isBlank(dockerHostEnv)) {
-                builder.withUri("unix:///var/run/docker.sock");
-            }
+    public static DockerClient build(Properties argProperties) {
+        properties = argProperties;
+        return build();
+    }
+
+    public static DockerClientConfigBuilder populateConfigBuilder(DockerClientConfigBuilder builder) {
+        builder.withVersion("");
+        builder.withDockerCertPath("");
 
-            DockerClientConfig config = builder.build();
-            dockerClient = DockerClientBuilder.getInstance(config).build();
+        dockerHostEnv = System.getenv("DOCKER_HOST");
+        dockerCertPathEnv = System.getenv("DOCKER_CERT_PATH");
+        dockerHostPrp = properties.getProperty("docker.io.url");
+        dockerCertPathPrp = properties.getProperty("docker.io.dockerCertPath");
+        dockerUri = "unix:///var/run/docker.sock";
+        
+        if (dockerHostEnv != null) {
+            dockerUri = dockerHostEnv;
+        } else if (dockerHostPrp != null) {
+            dockerUri = dockerHostPrp;
+            builder.withProperties(properties);
+        }
+        if (dockerCertPathPrp != null && new File(dockerCertPathPrp).exists()) {
+            dockerCertPath = dockerCertPathPrp;
+        }
+        if (dockerCertPathEnv != null && new File(dockerCertPathEnv).exists()) {
+            dockerCertPath = dockerCertPathEnv;
+        }
+        builder.withUri(dockerUri.replace("tcp", "https"));
```

Comment:
It seems that in case of setting the Docker host the properties take precedence while in case of cert path the environment variable takes precedence.